src/Shaders/ShadersInclude/pbrBlockSubSurface.fx

```diff
@@ -157,6 +160,23 @@ struct subSurfaceOutParams
         float thickness = vThicknessParam.y;
     #endif
 
+    #ifdef SS_INTENSITY_TEXTURE
+        #if defined(SS_USE_GLTF_THICKNESS_TEXTURE)
+            #ifdef SS_REFRACTION
+                refractionIntensity *= intensityMap.r;
+            #elif defined(SS_TRANSLUCENCY)
+                translucencyIntensity *= intensityMap.r;

```

Comment:
I guess we should be able to have both here ?

Reply:
Need to have a look...

Reply:
We would need to support another texture...

It does not feel quite right to add a new texture on our side each time glTF uses a channel for some intensity modulation in an extension. We could end up having to support dozens of textures in the end, and we would need to add some non trivial code / obfuscate shader code with even more #defines to optimize cases when some textures are actually the same, to save samplers...

The glTF specs should rather handle cases where several extensions are used together (transmission, volume, translucency) and enforce the fact that the "intensity" texture should be the same but each extension using a different channel.

What do you think? And @bghgary?

For the time being, the translucency extension is a draft, so I don't think we should make any change on our side just yet.

Reply:
For discussion:
* rename `intensityTexture` I have just added into `refractionIntensityTexture`. Use the green channel (red in the glTF case)
* add a new `translucencyIntensityTexture`. Use the blue channel (blue also in the glTF case)

We would optimize usage of sampler only when `refractionIntensityTexture` == `translucencyIntensityTexture` == `thicknessTexture`.

We would also need to have two new properties to sub surface, `useRefractionIntensityTexture` and `useTranslucencyIntensityTexture`. For the time being, if the texture is used, we assume it is always used for both the refraction and translucency cases.

Reply:
We had talked about doing something smarter in terms of sampler management. I think the idea is to specify what samplers are needed and do some kind of sampler indirection (maybe with macros or something) to make it so that we don't have to do this for each new textures that is added every time.